### PR TITLE
Updating lookupNamedIntrinsic to no longer assert for an unexpected hwintrinsic method.

### LINKED ISSUE #24323: Assertion failed 'strcmp(methodName, "get_IsSupported") == 0'
When using Windows/arm64 altjit (protononjit.dll) on Windows/x64 to generate PMI asm diffs, I see:
```
Assert failure(PID 7144 [0x00001be8], Thread: 10360 [0x2878]): Assertion failed 'strcmp(methodName, "get_IsSupported") == 0' in 'System.Text.ASCIIUtility:GetIndexOfFirstNonAsciiChar_Sse2(long,long):long' (IL size 1031)
    File: f:\gh\coreclr12\src\jit\importer.cpp Line: 4352
    Image: f:\gh\coreclr10\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root\corerun.exe
```

and I get no diffs.

Command used (from root of clone with build):
```
call jit-diff diff --pmi --diff --base --base_root f:\gh\coreclr12 --arch x64 --build Checked --altjit protononjit.dll --corelib
```

Related to https://github.com/dotnet/coreclr/issues/20379?

@tannergooding @CarolEidt 
cc @dotnet/jit-contrib

### PR DESCRIPTION
**Pull request by tannergooding**
Resolves #24323.

CC. @CarolEidt, @BruceForstall

### CODE REVIEW

File: src/jit/importer.cpp [hunk 1]
```diff
@@ -4347,11 +4347,14 @@ NamedIntrinsic Compiler::lookupNamedIntrinsic(CORINFO_METHOD_HANDLE method)
...
 #endif // !defined(_TARGET_XARCH_) && !defined(_TARGET_ARM64_)
-        else
+        else if (strcmp(methodName, "get_IsSupported") == 0)
         {
-            assert(strcmp(methodName, "get_IsSupported") == 0);
             return NI_IsSupported_False;
         }
+        else
+        {
+            return gtIsRecursiveCall(method) ? NI_Throw_PlatformNotSupportedException : NI_Illegal;
```
Comment (tannergooding) [inline src/jit/importer.cpp]: Another potential benefit of this is that we may not need both the `Isa.cs` and `Isa.PlatformNotSupported.cs` files. We could just have `Isa.cs` and those not supported by the current platform would hit this path (if someone tried to call it without the appropriate check).
* The `Isa.cs` and `Isa.PlatformNotSupported.cs` files are identical, except one is recursive and one explicitly throws.

**Comment by tannergooding** [comment]
jit-diff passed when run locally:
```
dotnet.exe ..\jitutils\bin\jit-diff.dll diff --pmi --diff --base --base_root D:\repos\coreclr --arch x64 --build Debug --altjit protononjit.dll --corelib                                                       Using --output D:\repos\coreclr\bin\diffs
Using --base D:\repos\coreclr\bin\Product\Windows_NT.x64.Debug
Using --diff D:\repos\coreclr\bin\Product\Windows_NT.x64.Debug
Using --core_root D:\repos\coreclr\bin\tests\Windows_NT.x64.Debug\Tests\Core_Root

Warning: it is best practice to use a Release build for --core_root, --crossgen, and --test_root.

Beginning PMI Diffs for System.Private.CoreLib.dll
- Finished 1/1 Base 1/1 Diff [535.8 sec]
Completed PMI Diffs for System.Private.CoreLib.dll in 535.82s
Diffs (if any) can be viewed by comparing: D:\repos\coreclr\bin\diffs\dasmset_6\base D:\repos\coreclr\bin\diffs\dasmset_6\diff
Analyzing diffs...
Found 1 files with textual diffs.
PMI Diffs for System.Private.CoreLib.dll for x64 protononjit.dll
Summary:
(Lower is better)
Total bytes of diff: 0 (0.00% of base)
0 total files with size differences (0 improved, 0 regressed), 1 unchanged.
0 total methods with size differences (0 improved, 0 regressed), 24321 unchanged.
1 files had text diffs but not size diffs.
System.Private.CoreLib.dasm had 132 diffs
Completed analysis in 4.98s
```

File: src/jit/importer.cpp (hunk 1 above)
Comment (CarolEidt) [reply src/jit/importer.cpp]: That makes sense, though that would require require all JITs on all targets to handle the intrinsics in this way, right?

Comment (tannergooding) [reply src/jit/importer.cpp]: Yes. It would require all runtimes, once the APIs are supported by that runtime, to consistently handle them.

Comment (tannergooding) [reply src/jit/importer.cpp]: (I think it is probably a post 3.0 cleanup that could happen, after discussing it with Mono/Xamarin and CoreRT/UWP; depending on their support plans for the APIs).

**Pull request merged**